### PR TITLE
PageTableDuplicates (2 sorries)

### PR DESCRIPTION
**Pull request by MitchellBuckley**
As title describes. Nothing very fancy here.

### CODE REVIEW

**Comment by MitchellBuckley** [comment]
Ready for review, but no rush.

File: proof/refine/ARM/PageTableDuplicates.thy [hunk 1]
```diff
@@ -2227,42 +2227,81 @@ lemma handleRecv_valid_duplicates'[wp]:
...
+  apply (clarsimp simp: ct_not_inQ_def valid_dom_schedule'_def ct_idle_or_in_cur_domain'_def
+                        tcb_in_cur_domain'_def valid_queues'_def valid_queues_def valid_bitmapQ_def
+                        bitmapQ_def valid_queues_no_bitmap_def bitmapQ_no_L2_orphans_def bitmapQ_no_L1_orphans_def)
+  done
+
+lemma updateTimeStamp_invs'[wp]:
+  "updateTimeStamp \<lbrace>invs'\<rbrace>"
+  unfolding updateTimeStamp_def
+  apply (wpsimp wp: dmo_invs'_simple)
+     apply (wpsimp simp: getCurrentTime_def no_irq_def)+
```
Comment (corlewis) [inline proof/refine/ARM/PageTableDuplicates.thy:2269]: Out of curiosity, this doesn't work with a single `wpsimp`?

File: proof/refine/ARM/PageTableDuplicates.thy [hunk 2]
```diff
@@ -2227,42 +2227,81 @@ lemma handleRecv_valid_duplicates'[wp]:
...
+  done
+
+lemma updateTimeStamp_sch_act_simple[wp]:
+  "updateTimeStamp \<lbrace>sch_act_simple\<rbrace>"
+  unfolding updateTimeStamp_def sch_act_simple_def
+  by (wpsimp wp: dmo_invs'_simple simp: setCurTime_def)
+
+crunches updateTimeStamp
+  for ksCurThread[wp]: "\<lambda>s. P (ksCurThread s)"
+  and st_tcb_at'[wp]: "\<lambda>s. st_tcb_at' P t s"
```
Comment (corlewis) [inline proof/refine/ARM/PageTableDuplicates.thy:2279]: nitpick: I'm guessing that this can be upgraded to `pred_tcb_at'`

**Review by corlewis — APPROVED** [review]
Nice!

**Review by mktnk3 — APPROVED** [review]
Good to see more sorries removed :)

**Pull request merged**